Description: Explicitly link against all needed libraries, otherwise
 unrelated packages will FTBFS.
Author: Andreas Moog <amoog@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/770743

Index: heartbeat-3.0.5/membership/ccm/Makefile.am
===================================================================
--- heartbeat-3.0.5.orig/membership/ccm/Makefile.am	2011-07-06 10:31:24.814619753 -0400
+++ heartbeat-3.0.5/membership/ccm/Makefile.am	2011-07-06 10:32:24.564707388 -0400
@@ -83,7 +83,8 @@
 
 libccmclient_la_SOURCES	= ccmlib_memapi.c ccmlib_eventapi.c ccmlib.h
 libccmclient_la_LDFLAGS	= -version-info 1:0:0
-
+libccmclient_la_LIBADD	= $(gliblib) \
+			  -lplumb
 
 ## library libclm.la
 libclmdir		= $(libdir)/@HB_PKG@
@@ -96,6 +97,9 @@
 ## Add -DPOSIX_THREADS to CFLAGS to compile a thread-safe version library
 libclm_la_CFLAGS	= -g #-DPOSIX_THREADS
 libclm_la_LDFLAGS	= -version-info 1:0:0
+libclm_la_LIBADD        = libccmclient.la \
+			  $(gliblib) \
+			  -lplumb
 
 clmtest_SOURCES		= clmtest.c
 clmtest_LDADD		= libclm.la libccmclient.la \
Index: heartbeat-3.0.5/lib/apphb/Makefile.am
===================================================================
--- heartbeat-3.0.5.orig/lib/apphb/Makefile.am	2011-07-06 10:31:24.824619761 -0400
+++ heartbeat-3.0.5/lib/apphb/Makefile.am	2011-07-06 10:33:08.984772528 -0400
@@ -26,6 +26,8 @@
 
 AM_CFLAGS		= @CFLAGS@
 
+gliblib			= @GLIBLIB@
+
 ## libraries
 
 lib_LTLIBRARIES		= libapphb.la
@@ -39,3 +41,5 @@
 endif
 
 libapphb_la_LDFLAGS	= -version-info 2:0:0 
+libapphb_la_LIBADD	= $(gliblib) \
+			  -lplumb
Index: heartbeat-3.0.5/lib/hbclient/Makefile.am
===================================================================
--- heartbeat-3.0.5.orig/lib/hbclient/Makefile.am	2011-07-06 10:31:24.834619773 -0400
+++ heartbeat-3.0.5/lib/hbclient/Makefile.am	2011-07-06 10:31:26.724622545 -0400
@@ -45,7 +45,9 @@
 
 libhbclient_la_SOURCES	= client_lib.c
 libhbclient_la_LDFLAGS	= -version-info 1:0:0
-libhbclient_la_LIBADD	= $(top_builddir)/replace/libreplace.la  
+libhbclient_la_LIBADD	= $(top_builddir)/replace/libreplace.la \
+			  $(gliblib) \
+			  -lplumb
 
 ## binary progs
 halib_PROGRAMS		= api_test
